File: Python/simple-factory/simple01/TVFactory.py
from HaierTV import HaierTV
from HisenseTV import HisenseTV

class TVFactory(object):
    @staticmethod
    def produceTV(brand):
        if brand.lower() == 'haier':
            print("正在生产一个Haier电视...")
            return HaierTV()
        elif brand.lower() == 'hisense':
            print("正在生产一个Hisense电视...")
            return HisenseTV()
        else:
            print(brand + "电视无法生产...\n")

File: Homework/Simple Factory/ch4/sample01/TVFactory.java
package ch4.sample01;

public class TVFactory
{
	public static TV produceTV(String brand) throws Exception
	{
		if(brand.equalsIgnoreCase("Haier"))
		{
			System.out.println("Producing Haier TV...");
			return new HaierTV();
		}
		else if(brand.equalsIgnoreCase("Hisense"))
		{
			System.out.println("Producing Hisense TV...");
			return new HisenseTV();
		}
		else
		{
			throw new Exception("Sorry! Can not produce this TV.");
		}
	}
}